de.metas.adempiere.adempiere/base/src/main/java/org/adempiere/bpartner/model/interceptor/C_BPartner.java

```diff
@@ -60,4 +70,28 @@ public void createBPartnerStatsRecord(@NonNull final I_C_BPartner bpartner)
 	{
 		Services.get(IBPartnerStatsDAO.class).retrieveBPartnerStats(bpartner);
 	}
+
+	@ModelChange(timings = { ModelValidator.TYPE_BEFORE_CHANGE}, ifColumnsChanged = {I_C_BPartner.COLUMNNAME_C_BP_Group_ID})
+	public void updateBPartnerStatsRecord(@NonNull final I_C_BPartner bpartner)
+	{
+		final IBPartnerStats bpartnerStats = Services.get(IBPartnerStatsDAO.class).retrieveBPartnerStats(bpartner);
+		final I_C_BP_Group bpGroup = bpartner.getC_BP_Group();
+		final String creditStatus = bpGroup.getSOCreditStatus();
+		if (!Check.isEmpty(creditStatus,true))
+		{
+			setSO_CreditStatus(bpartnerStats, creditStatus);
+
+			// make sure the status is correct
+			Services.get(IBPartnerStatisticsUpdater.class)
+			.updateBPartnerStatistics(Env.getCtx(), Collections.singleton(bpartner.getC_BPartner_ID()), ITrx.TRXNAME_None);
+
+		}
+	}
+
+	private void setSO_CreditStatus(@NonNull final IBPartnerStats bpartnerStats, @NonNull final String creditStatus)
```

Comment:
i don't see the benefit of this method.
imho it's introducing more confusion than making things more clear. When you read the name you don't know what it does... does it also save the record or only it sets that field?!?